src/components/HeaderWithBackButton/types.ts

```diff
@@ -0,0 +1,113 @@
+import {ReactElement} from 'react';
+import {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+import type {Action} from '@hooks/useSingleExecution';
+import type {StepCounterParams} from '@src/languages/types';
+import type {AnchorPosition} from '@src/styles';
+import type {PersonalDetails, Policy, Report} from '@src/types/onyx';
+import type ChildrenProps from '@src/types/utils/ChildrenProps';
+import type IconAsset from '@src/types/utils/IconAsset';
+
+type ThreeDotsMenuItems = {
+    /** An icon element displayed on the left side */
+    icon?: IconAsset;
+
+    /** Text label */
+    text: string;
+
+    /** A callback triggered when the item is selected */
+    onSelected: () => void;
+};
+
+type HeaderWithBackButtonProps = ChildrenProps & {
+    /** Title of the Header */
+    title?: string;
+
+    /** Subtitle of the header */
+    subtitle: string | ReactElement;
```

Comment:
Is this prop compulsory?

Reply:
@VickyStash I think this should be a ReactNode

```suggestion
    subtitle?: ReactNode
```

+ change in Header component:
```ts
type HeaderProps = {
    /** Title of the Header */
    title?: ReactNode;

    /** Subtitle of the header */
    subtitle?: ReactNode;

    /** Should we show the environment badge (dev/stg)?  */
    shouldShowEnvironmentBadge?: boolean;

    /** Additional text styles */
    textStyles?: StyleProp<TextStyle>;
};
```